nuxeo-drive-client/nxdrive/tests/test_engine_offline.py

```diff
@@ -0,0 +1,132 @@
+        5. Call original method to resume engine
+        4. Check the pair_state when engine goes online
+        '''
+
+        global server_online
+        server_online = False
+        # Wait for GetChangeSummary to call check_offline method
+        sleep(35)
+
+        # Create a folder inside user folder and a file inside the folder
+        self.local_client_1.make_folder(u'/', 'NetworkError')
+        self.local_client_1.make_file(u'/NetworkError', 'NetworkErrorFile.txt', content="test network failure")
+        # Wait for events to be handled
+        sleep(35)
+
+        # Check pair_state and error_count for the folder/file because it should not try to sync when engine is offline
+        test_folder = self.get_dao_state_from_engine_1('/NetworkError')
+        self.assertEqual(test_folder.pair_state, 'locally_created')
+        self.assertEqual(test_folder.error_count, 0)
+        test_file = self.get_dao_state_from_engine_1('/NetworkError/NetworkErrorFile.txt')
+        self.assertEqual(test_file.pair_state, 'locally_created')
+        self.assertEqual(test_file.error_count, 0)
+
+        # Stop mocking and call original method
+        global server_online
+        server_online = True
+        # Wait for queue_manager to process for the folder/file
+        sleep(60)
+
+        # pair_state should be 'synchronized' for the folder/file
+        test_folder = self.get_dao_state_from_engine_1('/NetworkError')
+        self.assertEqual(test_folder.pair_state, 'synchronized')
+        test_file = self.get_dao_state_from_engine_1('/NetworkError/NetworkErrorFile.txt')
+        self.assertEqual(test_file.pair_state, 'synchronized')
+
+    def test_with_manual_pause_resume(self):
+
+        '''
+        1. Pause the engine
+        2. Create a local folder inside user folder
+        3. Create a file inside the folder
+        4. Check the pair_state and error_count for the folder/files when engine is paused
+        5. Resume the engine
+        4. Check the pair_state when engine is resumed
+        '''
+
+        # Suspend the engine
+        self.engine_1.suspend()
+
+        # Create a folder inside user folder and a file inside the folder
+        self.local_client_1.make_folder(u'/', 'ManualPause')
+        self.local_client_1.make_file(u'/ManualPause', 'ManualPauseFile.txt', content="test manual pause and resume")
+        # Wait for events to be handled
+        sleep(35)
+
+        # queue manager will not process for the folder/file when engine is paused and
+        # database should not have entries for the folder/file
+        test_folder = self.get_dao_state_from_engine_1('/ManualPause')
+        self.assertIsNone(test_folder)
+        test_file = self.get_dao_state_from_engine_1('/ManualPause/ManualPauseFile.txt')
+        self.assertIsNone(test_file)
+
+        # Resume the engine
+        self.engine_1.resume()
+        # Wait for queue_manager to process the folder/file
+        sleep(60)
```

Comment:
a couple of seconds should be enough. The events are already queued and the resume just launches the processors right away.